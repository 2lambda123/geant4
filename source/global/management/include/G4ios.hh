// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: G4ios.hh,v 1.4 1999-11-23 15:00:04 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// ---------------------------------------------------------------
// GEANT 4 class header file
//
// G4ios.hh
//
// ---------------------------------------------------------------
#ifndef included_G4ios
#define included_G4ios

#if defined(OO_DDL_TRANSLATION)
/*
 * stdlib needs to be included before iostream.h
 * during oddlx runs to work around a parser
 * problem of AIX ooddlx v4.0.2 with some versions of
 * AIX system header files.
 */
#include <stdlib.h>
#endif

#include "g4std/iostream"
#include "G4Types.hh"

extern G4std::ostream G4cout;
extern G4std::ostream G4cerr;

#define G4cin G4std::cin
#define G4endl G4std::endl

#endif
