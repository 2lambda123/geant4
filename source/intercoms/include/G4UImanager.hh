//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UImanager.hh,v 1.12 2001-10-02 00:32:06 asaim Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UImanager_h
#define G4UImanager_h 1

#include "globals.hh"
#include "g4rw/ctoken.h"
//#include "g4rw/tvordvec.h"
#include "g4std/vector"
#include "g4std/fstream"
#include "G4VStateDependent.hh"
#include "G4UIcommandStatus.hh"
class G4UIcommandTree;
class G4UIcommand;
class G4UIsession;
class G4UIcontrolMessenger;
class G4UnitsMessenger;
class G4UIaliasList;

// class description:
//
//  This is a singlton class which controls the command manipulation
// and the user interface(s). The constructor of this class MUST NOT 
// invoked by the user.
//

class G4UImanager : public G4VStateDependent
{
  public: // with description
      static G4UImanager * GetUIpointer();
      //  A static method to get the pointer to the only existing object
      // of this class.

  protected:
      G4UImanager();
  public:
      ~G4UImanager();
  private:
      G4UImanager(const G4UImanager &right);
      const G4UImanager & operator=(const G4UImanager &right);
      int operator==(const G4UImanager &right) const;
      int operator!=(const G4UImanager &right) const;

  public: // with description
      G4String GetCurrentValues(const char * aCommand);
      //  This method returns a string which represents the current value(s)
      // of the parameter(s) of the specified command. Null string will be
      // returned if the given command is not defined or the command does
      // not support the GetCurrentValues() method.
      void AddNewCommand(G4UIcommand * newCommand);
      //  This method register a new command.
      void RemoveCommand(G4UIcommand * aCommand);
      //  This command remove the registered command. After invokation of this
      // command, that particular command cannot be applied.
      void ExecuteMacroFile(G4String fileName);
      //  A macro file defined by the argument will be read by G4UIbatch object.
      G4int ApplyCommand(const char * aCommand);
      G4int ApplyCommand(G4String aCommand);
      //  These two methods are identical. A command (and parameter(s)) given
      // by the method's argument will be applied. Zero will be returned in 
      // case the command is successfully executed. Positive non-zero value
      // will be returned if the command couldn't be executed. The meaning of
      // this non-zero value is the following.
      //   The returned number : xyy
      //        x00 : G4CommandStatus.hh enumeration
      //         yy : the problematic parameter (first found)
      void StoreHistory(G4String fileName="G4history.macro");
      void StoreHistory(G4bool historySwitch,
             G4String fileName="G4history.macro");
      //  The executed commands will be stored in the defined file. If 
      // "historySwitch" is false, saving will be suspended.
      void ListCommands(G4String direc);
      //  All commands registored under the given directory will be listed to
      // G4cout.
      void SetAlias(const char * aliasLine);
      void SetAlias(G4String aliasLine);
      //  Define an alias. The first word of "aliasLine" string is the
      // alias name and the remaining word(s) is(are) string value
      // to be aliased.
      void RemoveAlias(const char * aliasName);
      void RemoveAlias(G4String aliasName);
      //  Remove the defined alias.
      void ListAlias();
      //  Print all aliases.


  public:
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4double initialValue,G4double finalValue,G4double stepSize);
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4int initialValue,G4int finalValue,G4int stepSize);

      virtual G4bool Notify(G4ApplicationState requestedState);
      //  This method is exclusively invoked by G4StateManager and the user
      // must not use this method.
 
  private:
      void PauseSession(G4String msg);
      void CreateMessenger();
      G4UIcommandTree* FindDirectory(const char* dirName);

  public:
  // following three methods will be removed quite soon.
      void Interact();
      void Interact(const char * promptCharacters);
      void Interact(G4String promptCharacters);

  private:
      static G4UImanager * fUImanager;
      static G4bool fUImanagerHasBeenKilled;
      G4UIcommandTree * treeTop;
      G4UIsession * session;
      G4UIcontrolMessenger * UImessenger;
      G4UnitsMessenger * UnitsMessenger;
      G4String savedParameters;
      G4UIcommand * savedCommand;
      G4int verboseLevel;
      G4std::ofstream historyFile;
      G4bool saveHistory;
      G4std::vector<G4String> histVec;
      G4UIaliasList* aliasList;
      
      G4bool pauseAtBeginOfEvent;
      G4bool pauseAtEndOfEvent;


  public: // with description
      G4String GetCurrentStringValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    int parameterNumber=1, G4bool reGet=true);
      G4String GetCurrentStringValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    const char * aParameterName, G4bool reGet=true);
      //  These six methods returns the current value of a parameter of the
      // given command. For the first three methods, the ordering number of
      // the parameter (1 is the first parameter) can be given, whereas,
      // other three methods can give the parameter name.
      //  If "reGet" is true, actual request of returning the current value
      // will be sent to the corresponding messenger, while, if it is false,
      // the value stored in G4Umanager will be used. The later case is valid
      // for the sequential invokation for the same command.

      inline void SetPauseAtBeginOfEvent(G4bool vl)
      { pauseAtBeginOfEvent = vl; }
      inline G4bool GetPauseAtBeginOfEvent() const
      { return pauseAtBeginOfEvent; }
      inline void SetPauseAtEndOfEvent(G4bool vl)
      { pauseAtEndOfEvent = vl; }
      inline G4bool GetPauseAtEndOfEvent() const
      { return pauseAtEndOfEvent; }
      //  If the boolean flags are true, Pause() method of G4StateManager is invoked
      // at the very begining (before generating a G4Event object) or at the end of
      // each event. So that, in case a (G)UI session is defined, the user can interact.


  public:
      inline G4UIcommandTree * GetTree() const
      { return treeTop; }
      inline G4UIsession * GetSession() const
      { return session; }
  public: // with description
      inline void SetSession(G4UIsession *const value)
      { session = value; }
      //  This method defines the active (G)UI session.
     void SetCoutDestination(G4UIsession *const value);
     //  This method defines the destination of G4cout/G4cerr stream.
     // For usual cases, this method will be invoked by a concrete
     // (G)UI session class object and thus the user needs not to invoke
     // this method by him(her)self.

  public:
      inline void SetVerboseLevel(G4int val)
      { verboseLevel = val; }
      inline G4int GetVerboseLevel() const
      { return verboseLevel; }
      inline G4int GetNumberOfHistory() const
      { return histVec.size(); }
      inline G4String GetPreviousCommand(G4int i) const
      { 
        G4String st;
        if(i>=0 && i<G4int(histVec.size()))
        { st = histVec[i]; }
        return st;
      }

  // Old methods kept for backward compatibility
  //    inline G4UIcommandTree * GetTree() const
  //    { return treeTop; };
  //    inline G4UIsession * GetSession() const
  //    { return session; };
  //    inline void SetSession(G4UIsession *const  value)
  //    { session = value; };

};

#endif

