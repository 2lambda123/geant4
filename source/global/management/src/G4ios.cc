//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
//
// $Id$
//
// 
// --------------------------------------------------------------
//      GEANT 4 class implementation file
//
// G4ios.cc
//
// History 1998 Nov. 3 Masayasu Nagamatu

#include "G4ios.hh"
#include "G4strstreambuf.hh"

G4ThreadLocal G4strstreambuf *G4coutbuf_G4MT_TLS_ = 0;
G4ThreadLocal G4strstreambuf *G4cerrbuf_G4MT_TLS_ = 0;
G4ThreadLocal std::ostream *G4cout_G4MT_TLS_ = 0;
G4ThreadLocal std::ostream *G4cerr_G4MT_TLS_ = 0;
#define G4coutbuf (*G4coutbuf_G4MT_TLS_)
#define G4cerrbuf (*G4cerrbuf_G4MT_TLS_)
#define G4cout (*G4cout_G4MT_TLS_)
#define G4cerr (*G4cerr_G4MT_TLS_)

void G4iosInitialization()
{
  if (G4coutbuf_G4MT_TLS_ == 0) G4coutbuf_G4MT_TLS_ = new G4strstreambuf;
  if (G4cerrbuf_G4MT_TLS_ == 0) G4cerrbuf_G4MT_TLS_ = new G4strstreambuf;
  if (G4cout_G4MT_TLS_ == 0) G4cout_G4MT_TLS_ = new std::ostream(G4coutbuf_G4MT_TLS_);
  if (G4cerr_G4MT_TLS_ == 0) G4cerr_G4MT_TLS_ = new std::ostream(G4cerrbuf_G4MT_TLS_);
}

void G4iosFinalization()
{
  delete G4cout_G4MT_TLS_;
  delete G4cerr_G4MT_TLS_; 
  delete G4coutbuf_G4MT_TLS_;
  delete G4cerrbuf_G4MT_TLS_;
}

void __attribute__ ((constructor)) my_init(void)
{
  G4iosInitialization();
  //printf("G4ios is initialized\n");
}

void __attribute__ ((destructor)) my_fini(void)
{
  G4iosFinalization();
  //printf("G4ios is finalized\n");
}
