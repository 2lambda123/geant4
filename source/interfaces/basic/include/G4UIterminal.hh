//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UIterminal.hh,v 1.8 2003-06-16 16:55:57 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UIterminal_h
#define G4UIterminal_h 1

#include <fstream>
#include "G4UImanager.hh"
#include "G4VBasicShell.hh"
#include "G4VUIshell.hh"

//
//   Description:
//
//   This class inherits the class G4UIsession.
//   This is the class to use a character-terminal sesion.
//
//   Usage:  
//       G4UIsession* terminalSession = new G4UIterminal; 
//   or  G4UIsession* terminalSession = new G4UIterminal(new your-shell); 
//
//     A character-terminal session  "terminalSession" is instantiated.
//     G4cout stream is redirected by default to the constructed instance.
//
//   terminalSession-> SessionStart(); // "terminalSession" is started.
//   delete terminalSession;           // "terminalSession"  is deleted.
//
//
//   In default(no arguments are given), csh-like shell is instantiated.
//   If you want to use another shell (e.g. tcsh-like), you can give
//   your favorite shell in an argument of the constructor.
//
//   Which shell? / How to define your own shell?
//   Currently two kinds of shells,
//                   G4UIcsh / G4UItcsh
//   , are presented. 
//   They inherit the abstract base class, G4VUIshell.
//   In order to define your own shell, 
//     - Define your own shell class derived from G4VUIshell.
//     - Implement GetCommandLine() method (pure virtual).
//     - Add more functionality, if need.
//
//   For more detail, see source codes.
//

class G4UIterminal : public G4VBasicShell {
private:
  G4UImanager* UI;
  // shell
  G4VUIshell* shell;

  // program states
  G4bool iExit;
  G4bool iCont;

public:
  G4UIterminal(G4VUIshell* aShell=0);
  ~G4UIterminal();

  void SetPrompt(const G4String& prompt);

private:
  void ExecuteCommand(G4String aCommand);
  G4String GetCommand(const char* msg=0);
  G4bool GetHelpChoice(G4int& aInt);
  void ExitHelp();
  
public:
  // These methods are implementation of corresponding virtual methods
  // of G4UIsession class.
  virtual G4UIsession* SessionStart();  
  virtual void PauseSessionStart(G4String msg);
  virtual G4int ReceiveG4cout(G4String coutString);
  virtual G4int ReceiveG4cerr(G4String cerrString);

};

#endif

