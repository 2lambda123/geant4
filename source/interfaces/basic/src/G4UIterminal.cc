// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: G4UIterminal.cc,v 1.5 1999-11-15 10:39:44 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "G4UIterminal.hh"
#include "G4StateManager.hh"
#include "G4UIcommandTree.hh"
#include "G4UIcommand.hh"
#include "G4UIcommandStatus.hh"
#include "G4ios.hh"
#ifdef WIN32
#  include <Strstrea.h>
#else
#  include <strstream.h>
#endif

G4UIterminal::G4UIterminal()
{
  UI = G4UImanager::GetUIpointer();
  UI->SetSession(this);
  UI->SetCoutDestination(this);
  G4StateManager * statM = G4StateManager::GetStateManager();
  promptCharacter = statM->GetStateString(statM->GetCurrentState())
                    + G4String("> ");
  iExit = false;
  iCont = false;
}

G4UIterminal::~G4UIterminal() 
{ 
 if( G4UImanager::GetUIpointer() != 0)
  {
    UI->SetSession(NULL);
    UI->SetCoutDestination(NULL);
    //    G4cout << "Terminal session deleted" << endl;
  }
}

G4UIsession * G4UIterminal::SessionStart()
{
  iExit = true;
  G4StateManager * statM = G4StateManager::GetStateManager();
  promptCharacter = statM->GetStateString(statM->GetCurrentState())
                  + G4String("> ");
  G4String newCommand = GetCommand();
  while( iExit )
  {
    ExecuteCommand(newCommand);
    promptCharacter = statM->GetStateString(statM->GetCurrentState())
                    + G4String("> ");
    newCommand = GetCommand();
  }
  return NULL;
}

void G4UIterminal::PauseSessionStart(G4String msg)
{
  promptCharacter = msg;
  iCont = true;
  G4String newCommand = GetCommand();
  while( iCont )
  {
    ExecuteCommand(newCommand);
    newCommand = GetCommand();
  }
}

G4int G4UIterminal::ReceiveG4cout(G4String coutString)
{
  cout << coutString << flush;
  return 0;
}

G4int G4UIterminal::ReceiveG4cerr(G4String cerrString)
{
  cerr << cerrString << flush;
  return 0;
}

void G4UIterminal::ExecuteCommand(G4String aCommand)
{

  if(aCommand.length()<2) return;

  int commandStatus = UI->ApplyCommand(aCommand);
  switch(commandStatus)
  {
    case fCommandSucceeded:
      break;
    case fCommandNotFound:
      G4cerr << "command <" << aCommand << "> not found" << endl;
       if ( aCommand.index("@@") != G4std::string::npos) {
       G4cout << "@@G4UIterminal" << endl;
       }
      break;
    case fIllegalApplicationState:
      G4cerr << "illegal application state -- command refused" << endl;
      break;
    case fParameterOutOfRange:
    case fParameterUnreadable:
    case fParameterOutOfCandidates:
    default:
      G4cerr << "command refused (" << commandStatus << ")" << endl;
  }
}
G4String G4UIterminal::GetCommand()
{
  G4String newCommand;
  G4String nullString;
  while( 1 )
  {
    G4cout << promptCharacter;
    G4cout.flush();
    newCommand.readLine( cin, FALSE );
    if (!cin.good()) { cin.clear(); newCommand = nullString; iExit=false; break; }
    newCommand = newCommand.strip(G4String::leading);
    if( newCommand.length() < 1 ) { break; }

    while( newCommand(newCommand.length()-1) == '_' )
    {
      G4String newLine;
      newCommand.remove(newCommand.length()-1);
      newLine.readLine( cin );
      if (!cin.good()) { cin.clear(); newCommand = nullString; iExit=false; break; }
      newCommand.append(newLine);
    }

    G4String nC = newCommand.strip(G4String::leading);


    // -------------------- nC.toUpper();
    if( nC.length() < 1 ){ break; }
    if( nC(0) == '#' )
           { G4cout << nC << endl; }
    else if( nC == "ls" || nC(0,3) == "ls " )
           { ListDirectory( nC ); }
    else if( nC == "pwd" )
           { 
	     G4cout << "Current Working Directory : " 
		    << GetCurrentWorkingDirectory() << endl; 
	   }
    else if( nC == "cd" || nC(0,3) == "cd " )
           { ChangeDirectoryCommand( nC ); }
    else if( nC == "help" || nC(0,5) == "help " )
           { TerminalHelp( nC ); }
    else if( nC(0) == '?' )
           { ShowCurrent( nC ); }
    else if( nC == "hist" || nC == "history" )
           {
	     G4int nh = UI->GetNumberOfHistory();
	     for(int i=0;i<nh;i++)
	       { G4cout << i << ": " << UI->GetPreviousCommand(i) << endl; }
	   }
    else if( nC(0) == '!' )
           {
	     G4String ss = nC(1,nC.length()-1);
	     G4int vl;
	     const char* tt = ss;
	     istrstream is((char*)tt);
	     is >> vl;
	     G4int nh = UI->GetNumberOfHistory();
	     if(vl>=0 && vl<nh)
	       { 
		 newCommand = UI->GetPreviousCommand(vl); 
		 G4cout << newCommand << endl;
		 break;
	       }
	     else
	       { G4cerr << "history " << vl << " is not found." << endl; }
	   }
    else if( nC == "exit" )
          { 
	    if( iCont )
	      { 
		G4cout << "You are now processing RUN." << endl;
		G4cout << "Please abort it using \"/run/abort\" command first" << endl;
		G4cout << " and use \"continue\" command until the application" << endl;
		G4cout << " becomes to Idle." << endl;
	      }
	    else
	      {
		iExit = false;
		newCommand = nullString;
		break;
	      }
	  }
    else if( nC == "cont" || nC == "continue" )
          { 
	    iCont = false;
	    newCommand = nullString;
	    break;
	  }
    else
      { break; }
  }
  return ModifyToFullPathCommand(newCommand);
}

G4bool G4UIterminal::GetHelpChoice(G4int& aInt){
  cin >> aInt;
  if(!cin.good()){
    cin.clear();
    cin.ignore(30,'\n');
    return false;
  } 
  return true;
}
void G4UIterminal::ExitHelp(){
  // cin.flush();
  char temp[100];
  cin.getline( temp, 100 );
}




