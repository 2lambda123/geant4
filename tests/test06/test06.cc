//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: test06.cc,v 1.3 2001-07-11 10:09:39 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "Tst06DetectorConstruction.hh"
#include "Tst06RunAction.hh"
#include "Tst06PrimaryGeneratorAction.hh"
#include "Tst06PhysicsList.hh"
#include "Tst06SteppingAction.hh"

#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4RunManager.hh"

#include "G4ios.hh"

int main(int argc,char** argv) {

  // Set the default random engine to RanecuEngine
  RanecuEngine defaultEngine;
  HepRandom::setTheEngine(&defaultEngine);

  // Run manager
  G4RunManager * runManager = new G4RunManager;

  // UserInitialization classes
  runManager->SetUserInitialization(new Tst06DetectorConstruction);
  runManager->SetUserInitialization(new Tst06PhysicsList);

  // UserAction classes
  runManager->SetUserAction(new Tst06RunAction);
  runManager->SetUserAction(new Tst06PrimaryGeneratorAction);
  //runManager->SetUserAction(new Tst06SteppingAction);

  if(argc==1)
  {
    // G4UIterminal is a (dumb) terminal.
    G4UIsession* session = new G4UIterminal;
    session->SessionStart();
    delete session;
  }
  else
  {
    G4UImanager* UImanager = G4UImanager::GetUIpointer();
    G4String command = "/control/execute ";
    G4String fileName = argv[1];
    UImanager->ApplyCommand(command+fileName);
  }

  delete runManager;
  return 0;
}

