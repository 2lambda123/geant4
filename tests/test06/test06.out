**********************************************
 Geant4 version $Name: not supported by cvs2svn $
                                (19-May-2003)
             Copyright : Geant4 Collaboration
**********************************************
SimpleBox is now made of Pb
/run/verbose 2
/run/initialize
userDetector->Construct() start.
SLog is registered to the default region.
physicsList->Construct() start.

phot:  Total cross sections from Sandia parametrisation. 
physicsList->setCut() start.
Tst06PhysicsList::SetCuts:CutLength : 1 (mm)
/gun/direction 1 0 0
#
# gamma 500 MeV in Al
#
/mydet/SelectMaterial Al
SimpleBox is now made of Al
/gun/particle gamma
/gun/energy 500 MeV
/run/beamOn 100

compt:  Total cross sections from a parametrisation. Good description from 10 KeV to (100/Z) GeV. 
       Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 1 keV to 100 GeV in 80 bins. 

conv:  Total cross sections from a parametrisation. Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1.022 MeV to 100 GeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV  to 100 TeV in 100 bins. 

eIoni:  delta cross sections from Moller+Bhabha. Good description from 1 KeV to 100 GeV.
        delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 100 TeV in 100 bins.
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

eBrem:  Total cross sections from a NEW parametrisation based on the EEDL data library. 
 Good description from 1 KeV to 100 GeV.
        log scale extrapolation above 100 GeV 
        Gamma energy sampled from a parametrised formula.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 

annihil:  Total cross section from Heilter formula(annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV  to 100 TeV in 100 bins. 

hIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV  to 100 TeV in 100 bins. 

MuIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 
Start closing geometry.
G4GeometryManager::ReportVoxelStats -- Voxel Statistics

    Total memory consumed for geometry optimisation:   0 kByte
    Total CPU time elapsed for geometry optimisation: 0 seconds
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 100
  User=11s Real=13s Sys=1.6s
#
# electron 500 MeV in Air
#
/mydet/SelectMaterial Air
SimpleBox is now made of Air
/gun/particle e-
/gun/energy 500 MeV
/run/beamOn 100

compt:  Total cross sections from a parametrisation. Good description from 10 KeV to (100/Z) GeV. 
       Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 1 keV to 1e+02 GeV in 80 bins. 

conv:  Total cross sections from a parametrisation. Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1 MeV to 1e+02 GeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV  to 1e+02 TeV in 100 bins. 

eIoni:  delta cross sections from Moller+Bhabha. Good description from 1 KeV to 100 GeV.
        delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins.
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

eBrem:  Total cross sections from a NEW parametrisation based on the EEDL data library. 
 Good description from 1 KeV to 100 GeV.
        log scale extrapolation above 100 GeV 
        Gamma energy sampled from a parametrised formula.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins. 

annihil:  Total cross section from Heilter formula(annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV  to 1e+02 TeV in 100 bins. 

hIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1e+02 TeV in 100 bins. 
        Step function: finalRange(mm)= 1,  dRoverRange= 0.2

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 1e+02 eV  to 1e+02 TeV in 100 bins. 

MuIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1e+03 PeV in 150 bins. 
Start Run processing.
Run terminated.
Run Summary
  Number of events processed : 100
  User=0.46s Real=0.49s Sys=0.03s
#
G4 kernel has come to Quit state.
Deletion of G4 kernel class start.
UserDetectorConstruction deleted.
UserPhysicsList deleted.
UserRunAction deleted.
UserPrimaryGenerator deleted.
EventManager deleted.
UImanager deleted.
StateManager deleted.
RunManager is deleting.
